tzrec/models/match_model.py

```diff
@@ -181,18 +181,28 @@ def sim(
         user_emb: torch.Tensor,
         item_emb: torch.Tensor,
         neg_for_each_sample: bool = False,
+        is_hstu: bool = False,
```

Comment:
do not modify the sim, overwrite sim in hstu.py, and explain the logic in comments